src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -1471,7 +1471,8 @@ protected void GenerateFindFirstChar()
                 int setCharsCount = 0, charClassIndex = 0;
                 bool canUseIndexOf =
                     !_leadingCharClasses[0].CaseInsensitive &&
-                    (setCharsCount = RegexCharClass.GetSetChars(_leadingCharClasses[0].CharClass, setChars)) > 0;
+                    (setCharsCount = RegexCharClass.GetSetChars(_leadingCharClasses[0].CharClass, setChars)) > 0 &&
+                    !RegexCharClass.IsNegated(_leadingCharClasses[0].CharClass);
```

Comment:
Was this a bug that it wasn't handling negated before?

Reply:
No, GetSetChars was doing the IsNegated test.  But this change requires being able to use GetSetChars regardless of negation, so I moved the check out.